src/Mono.Android/Android.Runtime/JavaProxyThrowable.cs

```diff
@@ -1,23 +1,83 @@
 using System;
+using System.Diagnostics;
+using System.Reflection;
+
+using StackTraceElement = Java.Lang.StackTraceElement;
 
 namespace Android.Runtime {
 
-	class JavaProxyThrowable : Java.Lang.Error {
+	sealed class JavaProxyThrowable : Java.Lang.Error {
 
 		public  readonly Exception InnerException;
 
-		public JavaProxyThrowable (Exception innerException)
-			: base (GetDetailMessage (innerException))
+		JavaProxyThrowable (string message, Exception innerException)
+			: base (message)
 		{
 			InnerException  = innerException;
 		}
 
-		static string GetDetailMessage (Exception innerException)
+		public static JavaProxyThrowable Create (Exception innerException)
+		{
+			if (innerException == null) {
+				throw new ArgumentNullException (nameof (innerException));
+			}
+
+			// We prepend managed exception type to message since Java will see `JavaProxyThrowable` instead.
+			var proxy = new JavaProxyThrowable ($"[{innerException.GetType}]: {innerException.Message}", innerException);
```

Comment:
*This* is where ``[System.Func`1[System.Type]]`` is coming from!

Which *still* doesn't make sense, as that would mean that `innerException.GetType()` is `Func<Type>`, but `innerException` is an `Exception`, so… how is that *possible*?  I remain flummoxed.

Reply:
Oh!  It's because `()` is missing!  This should be `innerException.GetType()` (with parens), not `innerException.GetType`!  I'm kinda surprised that `innerException.GetType` even compiled; apparently the compiler was just implicitly converting it to a `Func<Type>`, which kinda makes sense?